sklearn/utils/_tags.py

```diff
@@ -274,6 +274,17 @@ def get_tags(estimator) -> Tags:
     try:
         tags = estimator.__sklearn_tags__()
     except AttributeError as exc:
+        # Check for common error cases to provide better error messages
+        estimator_name = getattr(estimator, "__name__", str(type(estimator).__name__))
+
+        # Case 1: User passed a class instead of an instance
+        if hasattr(estimator, "__name__") and hasattr(estimator, "__init__"):
+            raise TypeError(
+                f"Expected an estimator instance, but got class '{estimator_name}'. "
+                f"Did you mean to instantiate it? e.g., {estimator_name}()"
+            ) from exc
```

Comment:
This is not enough to differentiate a class from an instance of this class. Instead we should check the exception message for instance, like
``if "__sklearn_tags__() missing 1 required positional argument: 'self" is str(exc):'``
in a new `except` block for a `TypeError`.

(we could also check `type(estimator)` or `inspect.is_class(estimator)` but I prefer the check on the error message here)